starboard/shared/starboard/player/video_dmp_reader.cc

```diff
@@ -406,6 +406,74 @@ VideoDmpReader::Registry* VideoDmpReader::GetRegistry() {
   return &s_registry;
 }
 
+SequentialVideoDmpReader::SequentialVideoDmpReader(
+    const char* filename,
+    ReadOnDemandOptions read_on_demand_options)
+    : VideoDmpReader(filename, read_on_demand_options),
+      audio_sample_duration_(audio_duration() / number_of_audio_buffers()) {}
+
+SbPlayerSampleInfo SequentialVideoDmpReader::GetPlayerSampleInfo(
+    SbMediaType type,
+    size_t index) {
+  SbPlayerSampleInfo sample_info =
+      VideoDmpReader::GetPlayerSampleInfo(type, index);
+
+  if (type == kSbMediaTypeAudio) {
+    SB_DCHECK(index >= last_received_index_);
+    SB_DCHECK(sample_info.timestamp >= total_discarded_duration_);
+    sample_info.timestamp -= total_discarded_duration_;
+    last_received_index_ = index;
+  }
+
+  return sample_info;
+}
+
+SbPlayerSampleInfo SequentialVideoDmpReader::GetPlayerSampleInfo(
+    SbMediaType type,
+    size_t index,
+    int64_t discarded_duration_from_front,
+    int64_t discarded_duration_from_back) {
+  SB_DCHECK(type == kSbMediaTypeAudio);
+  SB_DCHECK(index >= last_received_index_);
+
+  if (index > last_received_index_) {
```

Comment:
I think the idea to add a SequentialVideoDmpReader is good. But we should have a better way to prevent discarded duration of same same buffer being counted multiple times. For example,
1. call GetPlayerSampleInfo(Audio, 0, any duration, any duration).
2. call GetPlayerSampleInfo(Audio, 1, any duration, any duration).
3. call GetPlayerSampleInfo(Audio, 0, any duration, any duration) <-- this won't work as expected.

If we can read each sample buffer only once without Reset(), we can consider to enforce it through class design.